lib/cloudant.js

```diff
@@ -493,14 +500,14 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   };
 
   if (id) {
-    self.updateAttributes(model, id, data, {}, function(err, docs) {
+    self.update(model, {id: id}, data, {}, function(err, docs) {
       if (err && err.statusCode !== 404) return cb(err);
       else if (err && err.statusCode === 404) {
         // strip out the `_rev` property before creation
         delete data._rev;
         self.create(model, data, {}, createHandler);
       } else {
-        return cb(err, self.fromDB(model, mo, docs), {isNewInstance: false});
+        return cb(err, docs[0], {isNewInstance: false});
```

Comment:
@ssh24 there is a question for [Line#514](https://github.com/strongloop/loopback-connector-cloudant/pull/141/files#diff-fa06ecf3249c63aabcbf7ff1ae8ec186R514), not this line:

IMO, If a user doesn't provide `id` but the data contains `_rev`, then cloudant should refuse to insert the doc, not we "strip out the `_rev` property before creation" instead.